File: Algorithms/199. Binary Tree Right Side View/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public List<Integer> rightSideView(TreeNode root) {
    	List<Integer> answer=new ArrayList<Integer>();
    	if (root==null) {
    		return answer;
    	}
    	List<TreeNode> list=new ArrayList<TreeNode>();
    	list.add(root);
    	int k=list.size();
    	while (k>0) {
			answer.add(list.get(list.size()-1).val);
			for (int i = 0; i < k; i++) {
				if (list.get(i).left!=null) {
					list.add(list.get(i).left);
				}
				if (list.get(i).right!=null) {
					list.add(list.get(i).right);
				}
				list.remove(i);
				i--;
				k--;
			}
			k=list.size();
		}
    	return answer;
    }
}

File: Algorithms/199. Binary Tree Right Side View/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ans=[]
        if not root:
            return ans
        treeList=[]
        treeList.append(root)
        ans.append(root.val)
        while len(treeList)!=0:
            newTreeList=[]
            test=None
            for tree in treeList:
                if tree.left:
                    newTreeList.append(tree.left)
                    test=tree.left.val
                if tree.right:
                    newTreeList.append(tree.right)
                    test=tree.right.val
            if test:
                ans.append(test)
            treeList=newTreeList
        return ans